spring-integration-core/src/test/kotlin/org/springframework/integration/dsl/KotlinDslTests.kt

```diff
@@ -386,6 +416,27 @@ class KotlinDslTests : TestApplicationContextAware {
 				}
 			}
 
+		@Bean
+		fun nullCheckWireTapChannel() = QueueChannel()
+
+		@Bean
+		fun nullChannelFlow(@Qualifier("nullCheckWireTapChannel") nullCheckWireTapChannel: MessageChannel) =
+			integrationFlow {
+				wireTap(nullCheckWireTapChannel)
+				nullChannel()
+			}
+
+		@Bean
+		fun nullChannelWithTransformFlow(
+			@Qualifier("nullCheckWireTapChannel") nullCheckWireTapChannel: MessageChannel) =
+			integrationFlow {
+				transformWith {
```

Comment:
Looks like simple `inline fun <reified P : Any> transform(crossinline function: (P) -> Any) {` would do exactly what we have here so far.
That `id` is out of use anyway.

Reply:
OK. 
I mean just this instead:
```
transform<String>{ it.uppercase() }
```
In other words:
```
transform<String>{ it.uppercase() }
wireTap(nullCheckWireTapChannel)
nullChannel()
```